environments/production/common/ecr.tf

```
@@ -0,0 +1,40 @@
+module "ecr" {
+  source      = "../../common/ecr/"
+  tags        = var.tags
+  environment = var.environment
+}
+
+resource "aws_ssm_parameter" "ecr_url" {
+  for_each    = module.ecr.repository_urls
+  name        = "/${replace(upper(each.key), "-", "_")}_ECR_URL"
+  description = "${title(each.key)} ECR repository URL."
+  type        = "SecureString"
+  value       = each.value
+  tags        = var.tags
+}
+
+resource "aws_ecr_repository_policy" "ecr_allow_staging_and_development" {
+  for_each   = module.ecr.repository_urls
+  repository = each.key
+
+  policy = jsonencode({
+    "Version" : "2008-10-17",
+    "Statement" : [
+      {
+        "Sid" : "allow_access_to_development_and_staging",
+        "Effect" : "Allow",
+        "Principal" : {
+          "AWS" : [
+            "arn:aws:iam::844815912454:root",
+            "arn:aws:iam::451934005581:root"
+          ]
+        },
+        "Action" : [
+          "ecr:BatchCheckLayerAvailability",
+          "ecr:BatchGetImage",
+          "ecr:GetDownloadUrlForLayer",
+          "ecr:ListImages"
+        ]
+    }]
```

```suggestion
    Statement = [
      {
        Sid    = "allow_access_to_development_and_staging"
        Effect = "Allow"
        Principal = {
          AWS = [
            "arn:aws:iam::844815912454:root",
            "arn:aws:iam::451934005581:root"
          ]
        },
        Action = [
          "ecr:BatchCheckLayerAvailability",
          "ecr:BatchGetImage",
          "ecr:GetDownloadUrlForLayer",
          "ecr:ListImages"
        ]
    }]
```

Just to be consistent, we can just write HCL here. We also don't need `Version` as it will default to the latest.